playbooks/roles/zowe/tasks/main.yml

```diff
@@ -36,6 +36,16 @@
     script_filename: ./zowe-install.sh
     script_parameters: "-i \"{{ zowe_root_dir }}\" -h \"{{ zowe_dataset_prefix }}\" -l \"{{ zowe_install_logs_dir }}\""
 
+# the use of zos_uss_user_profile here is a workaround, needed because otherwise JAVA_HOME is not set
+- name: Check fingerprint is correct
+  raw: >-
+    touch {{ zos_uss_user_profile }} && \
+    . {{ zos_uss_user_profile }} \
+    {{ zowe_environment_variable_overrides | default('') }} && \
+    cd "{{ zowe_root_dir }}" && \
+    bin/zowe-verify-authenticity.sh -l "{{ zowe_install_logs_dir }}"
+  ignore_errors: False
```

Comment:
Why is this in main as well as verify?

Reply:
It should be in main, not verify.

Reply:
Other way round; It should be in verify, not main.

Reply:
I've removed it from both now.